lib/HttpClient.js

```diff
@@ -268,8 +268,8 @@ function rawRequest(opts, cb) {
             req.getMetrics = function getMetrics() {
                 return metrics;
             };
+            opts.client._inflightRequests--;
```

Comment:
Should this be performed by a function that also checks (asserts?) that the number of in-flight requests is e.g. >= 0? I don't think we'd want to check for an upper bound.